src/ui/map.js

```diff
@@ -2761,6 +2784,48 @@ class Map extends Camera {
         return this;
     }
 
+    /**
+     * Update the average visible elevation by sampling terrain
+     *
+     * @returns {boolean} true if elevation has changed from the last sampling
+     */
+    _updateAverageElevation(timeStamp: number, ignoreTimeout: boolean=false): boolean {
+        if (timeStamp === undefined) throw new Error('undefined time!');
+        if (!this.painter.averageElevationNeedsEasing()) {
+            if (this.transform.averageElevation !== 0) {
+                this.transform.averageElevation = 0;
+                return true;
+            }
+            return false;
+        }
+
+        const timeoutElapsed = ignoreTimeout || timeStamp - this._averageElevationLastSampledAt > AVERAGE_ELEVATION_SAMPLING_INTERVAL;
+
+        if (timeoutElapsed && !this._averageElevation.isEasing(timeStamp)) {
+            this._averageElevationLastSampledAt = timeStamp;
+
+            const currentElevation = this.transform.averageElevation;
+            const newElevation = this.transform.sampleAverageElevation(this);
+            const elevationChange = Math.abs(currentElevation - newElevation);
+
+            if (elevationChange > AVERAGE_ELEVATION_CHANGE_THRESHOLD) {
+                this._averageElevation.easeTo(newElevation, timeStamp, 300);
+                return true;
+            } else if (elevationChange > 1e-8) {
```

Comment:
I believe it was actually chose carefully. The numbers should be chosen as large as possible to skip work while avoiding any undesirable behavior. The basic logic was:

1. changes larger than 10cm should be eased to (could maybe be 1m?)
2. changes less than 10cm can just be jumped to without easing
3. changes less than epsilon shouldn't even be applied at all

The trick then is: what's the right epsilon? I believe I've seen changes come back at ~1e-12. But realistically, this could be probably be 1e-3 (1mm) without any consequences.

Reply:
Makes sense!

Reply:
I've pulled this out as another constant, for clarity. Now it's `AVERAGE_ELEVATION_EASE_THRESHOLD` for changes which should be eased, and `AVERAGE_ELEVATION_CHANGE_THRESHOLD` for changes which should be applied at all.